src/Adapter/MSTest.TestAdapter/Execution/TestMethodRunner.cs

```diff
@@ -81,36 +84,34 @@ internal UnitTestResult[] Execute()
 
         UnitTestResult[]? result = null;
 
+        bool shouldRunClassCleanup = false;
+        bool shouldRunClassAndAssemblyCleanup = false;
+        bool runFailed = false;
+        using LogMessageListener logListener = new(_captureDebugTraces);
         try
         {
-            using (LogMessageListener logListener = new(_captureDebugTraces))
+            try
             {
-                try
-                {
-                    // Run the assembly and class Initialize methods if required.
-                    // Assembly or class initialize can throw exceptions in which case we need to ensure that we fail the test.
-                    _testMethodInfo.Parent.Parent.RunAssemblyInitialize(_testContext.Context);
-                    _testMethodInfo.Parent.RunClassInitialize(_testContext.Context);
-                }
-                finally
-                {
-                    initializationLogs = logListener.GetAndClearStandardOutput();
-                    initializationTrace = logListener.GetAndClearDebugTrace();
-                    initializationErrorLogs = logListener.GetAndClearStandardError();
-                    initializationTestContextMessages = _testContext.GetAndClearDiagnosticMessages();
-                }
+                // Run the assembly Initialize methods if required.
+                // Assembly initialize can throw exceptions in which case we need to ensure that we fail the test.
+                _testMethodInfo.Parent.Parent.RunAssemblyInitialize(_testContext.Context);
+            }
+            finally
+            {
+                initializationLogs = logListener.GetAndClearStandardOutput();
+                initializationTrace = logListener.GetAndClearDebugTrace();
+                initializationErrorLogs = logListener.GetAndClearStandardError();
+                initializationTestContextMessages = _testContext.GetAndClearDiagnosticMessages();
             }
-
-            // Listening to log messages when running the test method with its Test Initialize and cleanup later on in the stack.
-            // This allows us to differentiate logging when data driven methods are used.
-            result = RunTestMethod();
         }
         catch (TestFailedException ex)
         {
+            runFailed = true;
             result = [new UnitTestResult(ex)];
         }
         catch (Exception ex)
         {
+            runFailed = true;
             if (result == null || result.Length == 0)
```

Comment:
This logic is now obsolete. Result is never set from this path.

Reply:
no it comes here if assinit throws